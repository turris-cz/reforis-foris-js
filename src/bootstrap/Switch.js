/*
 * Copyright (c) 2020-2024 CZ.NIC z.s.p.o. (https://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from "react";

import PropTypes from "prop-types";
import { useUID } from "react-uid";

Switch.propTypes = {
    label: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.element,
        PropTypes.node,
        PropTypes.arrayOf(PropTypes.node),
    ]).isRequired,
    helpText: PropTypes.string,
    switchHeading: PropTypes.bool,
};

function Switch({ label, helpText, switchHeading, ...props }) {
    const uid = useUID();

    return (
        <div
            className={`form-check form-switch mb-3 ${
                switchHeading ? "d-flex align-items-center" : null
            }`.trim()}
        >
            <input
                type="checkbox"
                className={`form-check-input ${switchHeading ? "me-2" : ""}`.trim()}
                role="switch"
                id={uid}
                {...props}
            />
            <label className="form-check-label" htmlFor={uid}>
                {label}
            </label>
            {helpText && (
                <div className="form-text">
                    <small>{helpText}</small>
                </div>
            )}
        </div>
    );
}

export default Switch;
