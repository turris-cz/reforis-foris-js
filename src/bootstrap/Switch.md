Switch example:

```jsx
<Switch label="Enable Switch" helpText="Toggle that switch!" />
```
