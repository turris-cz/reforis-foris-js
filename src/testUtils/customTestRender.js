/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

/* eslint import/export: "off" */

import React from "react";
import { UIDReset } from "react-uid";
import { StaticRouter } from "react-router-dom";
import { render } from "@testing-library/react";
import PropTypes from "prop-types";

import { AlertContextMock } from "./alertContextMock";
import { CustomizationContextMock } from "./customizationContextMock";

Wrapper.propTypes = {
    children: PropTypes.oneOfType([
        PropTypes.arrayOf(PropTypes.node),
        PropTypes.node,
    ]),
};

function Wrapper({ children }) {
    return (
        <CustomizationContextMock>
            <AlertContextMock>
                <StaticRouter>
                    <UIDReset>{children}</UIDReset>
                </StaticRouter>
            </AlertContextMock>
        </CustomizationContextMock>
    );
}

const customTestRender = (ui, options) =>
    render(ui, { wrapper: Wrapper, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customTestRender as render };
