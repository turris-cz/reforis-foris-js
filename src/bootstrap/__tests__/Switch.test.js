/*
 * Copyright (C) 2020 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from "react";

import { render } from "customTestRender";

import Switch from "../Switch";

describe("<Switch/>", () => {
    it("Render switch", () => {
        const { container } = render(
            <Switch
                label="Test label"
                checked
                helpText="Some help text"
                onChange={() => {}}
            />
        );
        expect(container.firstChild).toMatchSnapshot();
    });

    it("Render uncheked switch", () => {
        const { container } = render(
            <Switch label="Test label" helpText="Some help text" />
        );
        expect(container.firstChild).toMatchSnapshot();
    });
});
