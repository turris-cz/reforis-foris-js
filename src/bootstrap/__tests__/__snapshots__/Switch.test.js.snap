// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Switch/> Render switch 1`] = `
<div
  class="form-check form-switch mb-3"
>
  <input
    checked=""
    class="form-check-input"
    id="1"
    role="switch"
    type="checkbox"
  />
  <label
    class="form-check-label"
    for="1"
  >
    Test label
  </label>
  <div
    class="form-text"
  >
    <small>
      Some help text
    </small>
  </div>
</div>
`;

exports[`<Switch/> Render uncheked switch 1`] = `
<div
  class="form-check form-switch mb-3"
>
  <input
    class="form-check-input"
    id="1"
    role="switch"
    type="checkbox"
  />
  <label
    class="form-check-label"
    for="1"
  >
    Test label
  </label>
  <div
    class="form-text"
  >
    <small>
      Some help text
    </small>
  </div>
</div>
`;
